### PR TITLE
Remove unused-parameter warnings, round 2 (5 of 19)

### PR DESCRIPTION
**Pull request by vjpai**
The last round of unused-parameter fixes was generated based on unused parameters in a DEBUG build on one platform (Linux? Mac? I forget). As a result, it didn't complain about all the debug-only parameters and also only saw usage from one platform. It also had other random omissions. This round has been run on both Linux and Mac (sorry, Windows, we'lll look at you soon).

This round of unused-parameter warnings was more complex in some cases because it wasn't just trivial commenting of parameter names. In some cases, function parameters were changed; in some others, (void) expressions were added to convince the compiler that a parameter was indeed being used.

### CODE REVIEW

File: src/core/ext/transport/chttp2/transport/hpack_parser.cc [hunk 1]
```diff
@@ -811,11 +811,11 @@ static grpc_error* parse_indexed_field_x(grpc_chttp2_hpack_parser* p,
    cached metadata corresponds with the index we are examining. */
 static grpc_mdelem get_precomputed_md_for_idx(grpc_chttp2_hpack_parser* p) {
   GPR_DEBUG_ASSERT(p->md_for_index.payload != 0);
-  GPR_DEBUG_ASSERT(static_cast<int64_t>(p->index) == p->precomputed_md_index);
   grpc_mdelem md = p->md_for_index;
   GPR_DEBUG_ASSERT(!GRPC_MDISNULL(md)); /* handled in string parsing */
   p->md_for_index.payload = 0; /* Invalidate cached md when index changes. */
 #ifndef NDEBUG
+  GPR_DEBUG_ASSERT(static_cast<int64_t>(p->index) == p->precomputed_md_index);
```
Comment (AspirinSJL) [inline src/core/ext/transport/chttp2/transport/hpack_parser.cc]: This change seems unrelated to this PR?

File: src/core/ext/transport/chttp2/transport/hpack_parser.cc [hunk 2]
```diff
@@ -1541,9 +1543,9 @@ static bool is_binary_literal_header(grpc_chttp2_hpack_parser* p) {
 static void set_precomputed_md_idx(grpc_chttp2_hpack_parser* p,
                                    grpc_mdelem md) {
   GPR_DEBUG_ASSERT(p->md_for_index.payload == 0);
-  GPR_DEBUG_ASSERT(p->precomputed_md_index == -1);
   p->md_for_index = md;
 #ifndef NDEBUG
+  GPR_DEBUG_ASSERT(p->precomputed_md_index == -1);
```
Comment (AspirinSJL) [inline src/core/ext/transport/chttp2/transport/hpack_parser.cc]: Same here.

File: src/core/ext/transport/chttp2/transport/hpack_parser.cc (hunk 1 above)
Comment (vjpai) [reply src/core/ext/transport/chttp2/transport/hpack_parser.cc]: It moved down a few lines so I could keep it within a common `#ifndef DEBUG` that was already there rather than adding a new one. If you'd prefer, I can keep it in its original place and just put in a new `#ifndef` block

File: src/core/ext/transport/chttp2/transport/hpack_parser.cc (hunk 2 above)
Comment (vjpai) [reply src/core/ext/transport/chttp2/transport/hpack_parser.cc]: Same response here also, and same offer.

File: src/core/ext/transport/chttp2/transport/hpack_parser.cc (hunk 1 above)
Comment (AspirinSJL) [reply src/core/ext/transport/chttp2/transport/hpack_parser.cc]: But how is it related to this PR?

Comment (vjpai) [reply src/core/ext/transport/chttp2/transport/hpack_parser.cc]: Oh you're right; I'll revert this change. This got in because of a change I had attempted earlier which ended up not making sense.

File: src/core/ext/transport/chttp2/transport/hpack_parser.cc (hunk 2 above)
Comment (vjpai) [reply src/core/ext/transport/chttp2/transport/hpack_parser.cc]: I'll revert this change also.
